xGame/client/src/Components/MatchSelect/MatchSelect.js

```diff
@@ -1,14 +1,121 @@
 import React from "react";
 import { Collapse } from "reactstrap";
 import PropTypes from "prop-types";
+import Button from '@material-ui/core/Button';
+import TextField from '@material-ui/core/TextField';
+import Dialog from '@material-ui/core/Dialog';
+import DialogContent from '@material-ui/core/DialogContent';
+import DialogContentText from '@material-ui/core/DialogContentText';
+import DialogTitle from '@material-ui/core/DialogTitle';
+import { Row, Col } from "reactstrap";
+import { Formik, Field } from "formik";
 
 const MatchSelect = (props) => {
   const { isOpen } = props;
+  const [open, setOpen] = React.useState(false);
+
+  const handleClickOpen = () => {
+      setOpen(true);
+    };
+
+  const handleClose = () => {
+      setOpen(false);
+  };
+
```

Comment:
I'd suggest combining these two functions into one, called "toggleShowMatchCreation", which just does setOpen(!open) (or setShowMatchCreation(!showMatchCreation) if you use my suggestion from the comment above)